release/nightly_tests/dataset/benchmark.py

```diff
@@ -50,32 +49,108 @@ def __init__(self, name):
         self.result = {}
         print(f"Running benchmark: {name}")
 
-    def run(self, name: str, fn: Callable[..., Dataset], **fn_run_args):
+    def run_materialize_ds(
+        self,
+        name: str,
+        fn: Callable[..., Dataset],
+        *fn_args,
+        **fn_kwargs,
+    ):
+        """Run a benchmark on materializing a Ray Dataset. ``fn`` is expected to
+        return the Dataset which is to be materialized. Runtime and throughput
+        are automatically calculated and reported."""
+
         gc.collect()
 
         print(f"Running case: {name}")
         start_time = time.perf_counter()
-        output_ds = fn(**fn_run_args)
+        output_ds = fn(*fn_args, **fn_kwargs)
         output_ds.materialize()
         duration = time.perf_counter() - start_time
 
         # TODO(chengsu): Record more metrics based on dataset stats.
-        self.result[name] = {BenchmarkMetric.RUNTIME.value: duration}
+        self.result[name] = {
+            BenchmarkMetric.RUNTIME.value: duration,
+            BenchmarkMetric.THROUGHPUT.value: output_ds.count() / duration,
+        }
         print(f"Result of case {name}: {self.result[name]}")
 
-    def run_fn(self, name: str, fn: Callable[..., Any], **fn_run_args):
+    def run_iterate_ds(
+        self,
+        name: str,
+        dataset: Any,
+    ):
+        """Run a benchmark iterating over a dataset. Runtime and throughput
+        are automatically calculated and reported. Supported dataset types are:
+        - Ray Dataset (`ray.data.Dataset`)
+        - iterator over Ray Dataset (`ray.data.iterator._IterableFromIterator` from
+            `.iter_batches()`,`.iter_torch_batches()`, `.iter_tf_batches()`)
+        - Torch DataLoader (`torch.utils.data.DataLoader`)
+        - TensorFlow Dataset (`tf.data.Dataset`)
+        """
+        # Import TF/Torch within this method, as not all benchmarks
+        # will use/install these libraries.
+        import tensorflow as tf
+        import torch
+
+        gc.collect()
+
+        print(f"Running case: {name}")
+        start_time = time.perf_counter()
+        record_count = 0
+        ds_iterator = iter(dataset)
+        for first_batch in ds_iterator:
```

Comment:
why is it named "first_batch", instead of just "batch"?

Reply:
ah thanks for the catch, leftover bad naming from previous implementation.